### PR TITLE
refactor: ito new proxy contract

### PR DESCRIPTION
**Pull request by zhouhanseng**
1. multi contract
2. bug fix

### CODE REVIEW

**Review by guanbinrui — DISMISSED** [review]

File: pnpm-lock.yaml [hunk 1]
```diff
@@ -418,7 +418,7 @@ importers:
       '@openzeppelin/contracts': 3.4.1
       '@popperjs/core': 2.9.2
       '@servie/events': 3.0.0
-      '@snapshot-labs/snapshot.js': github.com/snapshot-labs/snapshot.js/f32bf0f5b4a27a23b0db3c6fdaced1abab25f55a
+      '@snapshot-labs/snapshot.js': github.com/snapshot-labs/snapshot.js/e2a6ac28560f22030104b39bb983294a0a3cb73d
```
Comment (septs) [inline pnpm-lock.yaml:421]: @Jack-Works i think need lock snapshot.js version
CC @guanbinrui

Comment (septs) [reply pnpm-lock.yaml:421]: @zhouhanseng suggest, revert this pnpm-lock changed

Comment (guanbinrui) [reply pnpm-lock.yaml:421]: It's locked by commit hash.

Comment (septs) [reply pnpm-lock.yaml:421]: https://github.com/DimensionDev/Maskbook/blob/7fe73abd542ae7271fcd3c16cac747efcb6d02d9/packages/maskbook/package.json#L35

Comment (septs) [reply pnpm-lock.yaml:421]: https://github.com/DimensionDev/Maskbook/pull/3467#pullrequestreview-689599022

Comment (guanbinrui) [reply pnpm-lock.yaml:421]: I think this pkg should not be updated in this PR. We need to lock the version of `@snapshot-labs/snapshot.js` in the `package.json` file.

Comment (septs) [reply pnpm-lock.yaml:421]: https://github.com/DimensionDev/Maskbook/pull/3490
plz review lock snapshot.js pr

**Review by septs — APPROVED** [review]

File: packages/maskbook/src/plugins/ITO/UI/ClaimAllDialog.tsx [hunk 2]
```diff
@@ -105,25 +120,69 @@ export function ClaimAllDialog(props: ClaimAllDialogProps) {
...
+        const claimableTokens = swappedTokensOld!.filter((t) => t.isClaimable)
+        const summary =
+            'Claim ' +
+            new Intl.ListFormat('en').format(
+                claimableTokens.map((t) => formatBalance(t.amount, t.token.decimals) + ' ' + t.token.symbol),
+            )
+        setClaimTransactionDialog({
+            open: true,
+            state: claimStateOld,
+            title: 'Claim Token',
```
Comment (guanbinrui) [inline packages/maskbook/src/plugins/ITO/UI/ClaimAllDialog.tsx:181]: `i18N`

**Review by guanbinrui — DISMISSED** [review]

**Review by guanbinrui — APPROVED** [review]

**Pull request merged**